conformance/runner.js

```diff
@@ -105,14 +120,87 @@ function convertCommit(commitRequest) {
   return deepCopy;
 }
 
-/** Converts a Protobuf Precondition to the expected user input. */
+/** Converts a Position in Proto3 JSON to Protobuf JS format. */
+function convertPosition(position) {
+  const values = [];
+  for (let value of position.values) {
+    values.push(convert.valueFromJson(value));
+  }
+  position.values = values;
+  return position;
+}
+
+/** Converts a RunQueryRequest in Proto3 JSON to Protobuf JS format. */
+function convertQuery(queryRequest) {
+  const deepCopy = JSON.parse(JSON.stringify(queryRequest));
+  if (deepCopy.where && deepCopy.where.fieldFilter) {
+    deepCopy.where.fieldFilter.value = convert.valueFromJson(
+      deepCopy.where.fieldFilter.value
+    );
+  }
+  if (deepCopy.where && deepCopy.where.compositeFilter) {
+    for (let filter of deepCopy.where.compositeFilter.filters) {
+      filter.fieldFilter.value = convert.valueFromJson(
+        filter.fieldFilter.value
+      );
+    }
+  }
+  if (deepCopy.startAt) {
+    deepCopy.startAt = convertPosition(deepCopy.startAt);
+  }
+  if (deepCopy.endAt) {
+    deepCopy.endAt = convertPosition(deepCopy.endAt);
+  }
+  return deepCopy;
+}
+
+/** Converts a `Precondition` to its API counterpart. */
 function convertPrecondition(precondition) {
```

Comment:
We have some convert*() methods that convert between protobuf formats, and some that convert between API format and proto format. This seems confusing and potentially error-prone. Does it make sense to have an apiConverter "namespace" and protoConverter "namespace" to clarify what kind of conversions these methods do?

Reply:
It was actually rather confusing when I went through and update the comments. I created two namespaces (one for the input and one for the Protobuf conversion) and translated what was in the comments into code.